engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/internal/BaseWindowedFloatUpdateByOperator.java

```diff
@@ -0,0 +1,180 @@
+package io.deephaven.engine.table.impl.updateby.internal;
+
+import io.deephaven.api.updateby.OperationControl;
+import io.deephaven.chunk.Chunk;
+import io.deephaven.chunk.IntChunk;
+import io.deephaven.chunk.LongChunk;
+import io.deephaven.chunk.WritableFloatChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.table.*;
+import io.deephaven.engine.table.impl.sources.FloatArraySource;
+import io.deephaven.engine.table.impl.sources.FloatSparseArraySource;
+import io.deephaven.engine.table.impl.sources.WritableRedirectedColumnSource;
+import io.deephaven.engine.table.impl.updateby.UpdateByWindowedOperator;
+import io.deephaven.engine.table.impl.util.RowRedirection;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.Map;
+
+import static io.deephaven.engine.rowset.RowSequence.NULL_ROW_KEY;
+import static io.deephaven.util.QueryConstants.NULL_FLOAT;
+
+public abstract class BaseWindowedFloatUpdateByOperator extends UpdateByWindowedOperator {
+    protected final WritableColumnSource<Float> outputSource;
+    protected final WritableColumnSource<Float> maybeInnerSource;
+
+    public float curVal = NULL_FLOAT;
```

Comment:
This belongs on the context, right? Seems like an argument for replicating from char.

Reply:
Now replicating from Char version